src/kernels/xform_bidirect_winograd_code.inc

```diff
@@ -1082,6 +1093,7 @@ skip_lds_init:
         i = 0
         .rept \num
             .if \elem_size == 2
+                v_cvt_f16_f32 v[\vsrc + i], v[\vsrc + i]
```

Comment:
Convertions should be decoupled from loads and stores. The best way is to add macro for that:
```
.macro v_cvt dst, src, dst_type, src_type
    .if \dst_type == \src_type
        // no op
    .elseif \dst_type == TYPE_FP32 && \src_type == TYPE_FP16
        v_cvt_f32_f16 \dst, \src
    .elseif \dst_type == TYPE_FP32 && \src_type == TYPE_BFP16
        v_lshlrev_b32 \dst, 16, \src
    .elseif \dst_type == TYPE_FP16 && \src_type == TYPE_FP32
        v_cvt_f16_f32 \dst, \src
    .elseif \dst_type == TYPE_BFP16 && \src_type == TYPE_FP32
        v_lshrrev_b32 \dst, 16, \src // TODO: add rounding, etc
    .else
        assert(0) // unknown type or convertion is not implemented
    .endif
.endm
.macro v_cvt_array v, n, dst_type, src_type
    i = 0 // TODO: switch i to local scope
    .rept \n
        v_cvt v[\v + i], v[\v + i], \dst_type, \src_type
        i = i + 1
    .endr
.endm
```